### PR TITLE
update connection way to improve slow connection issue

### PR DESCRIPTION
**Pull request by dwlee**
To solve following issue
https://github.com/robotics-in-concert/rocon_tools/blob/indigo/rocon_python_comms/src/rocon_python_comms/services.py#L48

related PR
https://github.com/robotics-in-concert/rocon_tools/pull/79

### CODE REVIEW

File: rocon_remocon/src/rocon_remocon/interactive_client.py [hunk 1]
```diff
@@ -104,9 +104,21 @@ def _connect(self, rocon_master_name="", ros_master_uri="http://localhost:11311"
...
         rospy.init_node(self.name, disable_signals=True)
         try:
-            get_interactions_service_name = rocon_python_comms.find_service('rocon_interaction_msgs/GetInteractions', timeout=rospy.rostime.Duration(5.0), unique=True)
-            get_roles_service_name = rocon_python_comms.find_service('rocon_interaction_msgs/GetRoles', timeout=rospy.rostime.Duration(5.0), unique=True)
-            request_interaction_service_name = rocon_python_comms.find_service('rocon_interaction_msgs/RequestInteraction', timeout=rospy.rostime.Duration(5.0), unique=True)
+            console.logdebug("Interactive Client : Get interactions service Handle")
+            interactions_namespace = rocon_python_comms.find_namespace('get_interactions', 'rocon_interaction_msgs/GetInteractions')
+
+            remocon_services = {}
+            remocon_services['get_interactions'] = interactions_namespace + '/' + 'get_interactions'
```
Comment (jihoonl) [inline rocon_remocon/src/rocon_remocon/interactive_client.py]: How about making `get_interactions`, `get_roles`, and `request_interaction` as class constant value?

Comment (stonier) [inline rocon_remocon/src/rocon_remocon/interactive_client.py]: Only necessary really if you use it in more than one place.

File: rocon_remocon/src/rocon_remocon/interactive_client.py [hunk 2]
```diff
@@ -104,9 +104,21 @@ def _connect(self, rocon_master_name="", ros_master_uri="http://localhost:11311"
...
+            remocon_services['get_roles'] = interactions_namespace + '/' + 'get_roles'
+            remocon_services['request_interaction'] = interactions_namespace + '/' + 'request_interaction'
+
+            for service_name in remocon_services.keys():
+                if not rocon_python_comms.is_valid_service(remocon_services[service_name]):
+                    raise rocon_python_comms.NotFoundException("'%s' service is not validated" % service_name)
+        except rocon_python_comms.MultipleFoundException as e:
+            message = "find multiple interactions' publications and services [%s]. Please check interactions remap rules" % str(e)
```
Comment (stonier) [inline rocon_remocon/src/rocon_remocon/interactive_client.py]: This should filter by the message type before issuing the warning. If the type is `rocon_interaction_msgs/GetInteractions.msg` then you can probably reduce to just 1 solution. Then issue warning if there are still multiple.

Comment (stonier) [inline rocon_remocon/src/rocon_remocon/interactive_client.py]: @jihoonl made a good comment too....this should do more than just warn...it should gracefully exit with the error message when it fails.

**Comment by jihoonl** [comment]
I am making changes to proceed quickly.

File: rocon_remocon/src/rocon_remocon/interactive_client.py (hunk 1 above)
Comment (jihoonl) [inline rocon_remocon/src/rocon_remocon/interactive_client.py]: Ok. NM

File: rocon_remocon/src/rocon_remocon/interactive_client.py (hunk 2 above)
Comment (jihoonl) [inline rocon_remocon/src/rocon_remocon/interactive_client.py]: FYI, `find_service_namespace` is already checking service type. So it did not need second filter to reduce namespace into 1. If it is multiple, they have same `service_name` and `service_type`. Thus, it should throw an error.

**Comment by jihoonl** [comment]
FYI, I have improved [find_service_namespace](https://github.com/robotics-in-concert/rocon_tools/blob/indigo/rocon_python_comms/src/rocon_python_comms/namespace.py#L36) function.

**Pull request merged**